pkg/store/egressbatchstore/fsstore.go

```diff
@@ -0,0 +1,169 @@
+package egressbatchstore
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2/blockstore"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-multihash"
+	"github.com/storacha/go-libstoracha/capabilities/space/content"
+	"github.com/storacha/go-ucanto/core/receipt"
+	fdm "github.com/storacha/go-ucanto/core/result/failure/datamodel"
+)
+
+const (
+	// DefaultBatchSize is the default maximum size of a receipt batch in bytes.
+	DefaultBatchSize = 100 * 1024 * 1024 // 100MiB
+
+	// currentBatchName is the name of the current batch file.
+	currentBatchName = "egress.car"
+
+	// batchFilePrefix is the prefix for completed batch files.
+	batchFilePrefix = "egress."
+	batchFileSuffix = ".car"
+)
+
+var _ EgressBatchStore = (*fsBatchStore)(nil)
```

Comment:
I am not convinced by the name - to me a store is something you put stuff in that you can also get stuff back out of. This is an append only log/logger - you cannot/do not want to retrieve items from it.

Reply:
actually, we will want to retrieve items from it when implementing the endpoint from which receipt batches will be fetched by the egress tracking service during consolidation, so I think it qualifies as a store. But happy to think of a different name if that argument still feels unconvincing.